ocf_data_sampler/config/model.py

```diff
@@ -212,16 +212,16 @@ class NWP(TimeWindowMixin, DropoutMixin, SpatialWindowMixin, NormalisationConsta
         " the maximum forecast horizon of the NWP and the requested forecast length.",
     )
 
-    @field_validator("accum_channels")
-    def validate_accum_channels_subset(cls, v: list[str], values: ValidationInfo) -> list[str]:
+    @model_validator(mode="after")
+    def validate_accum_channels_subset(self) -> "NWP":
         """Validate accum_channels is subset of channels."""
-        channels = values.data.get("channels", [])
-        invalid_channels = set(v) - set(channels)
+        invalid_channels = set(self.accum_channels) - set(self.channels)
         if invalid_channels:
             raise ValueError(
-                f"accum_channels contains channels not present in 'channels': {invalid_channels}",
+                f"NWP provider '{self.provider}': accum_channels contains "
+                f"invalid channels: {invalid_channels}",
```

Comment:
```suggestion
                f"NWP provider '{self.provider}': all values in 'accum_channels' should "
                f"be present in 'channels'. Extra values found: {invalid_channels}",
```

I'm nitpicking but I think this is clearer, as technically the channels are not invalid, they should just be present in channels also, and this tells you how to fix it.

Great job adding the provider in! Looks good now